docs/libvirt-node-networking.md

```diff
@@ -0,0 +1,298 @@
+# Layer Two Connectivity for nested k8s Nodes
+
+The purpose of this proposal, is to allow connecting VMs to the node network,
+in contrast to
+[docs/libvirt-pod-networking.md](docs/libvirt-pod-networking.md), which allows
+VMs to be connected to the Kubernetes Overlay network. This allows VMs started
+by KubeVirt, to act as full k8s nodes themselves. Inside the VMs, the overlay
+network provider used in the k8s installation will work as expected and the
+nested node looks like an ordinary node from networking perspective.
+
+There are three main points to consider, to achive this:
+
+ 1) How and where to specify what the master host network device to connect to
+    is.
+ 2) How can the nested nodes and the real node communicate with each other.
+ 3) How to bring a device in position, so that libvirt can use it.
+ 
+ Further the proposal focuses on two network topologies:
+ 
+ 1) The master host network device is **not** a **bridge**
+ 2) The master host network device is a **bridge**.
+ 
+ In both cases, the goal is to keep the host network as much as it is. Thus the
+ most optimal solution only adds network related object (device, routes, ...),
+ but does not modify existing configurations.
+
+ For both network topologies, the existence of a DHCP service for the nodes
+ network is required.
+
+## How to specify the host network device
+
+Right now, adding an extra network type `NodeNetwork` will be added. It does
+not take any extra arguments. On the node, the CNI plugin which creates a new
+endpoint for the device will get the node device name passed in by
+virt-handler. Virt-handler, which has to invoke the plugin from the host
```

Comment:
Why is CNI and virt-handler mentioned here, why is it relevant?
Wasn't this about how the node ip is determined?

Also: They are mentioned but I don't really understand what is happening.